emperor/support_files/js/view-controller.js

```diff
@@ -353,12 +373,53 @@ define([
       columns.unshift(options.slickGridColumn);
     }
 
+    var dataView = new Slick.Data.DataView(), searchString = '';
+
     /**
      * @type {Slick.Grid}
      * Container that lists the metadata categories described under the
      * metadata column and the attribute that can be modified.
      */
-    this.bodyGrid = new Slick.Grid(this.$gridDiv, [], columns, gridOptions);
+    this.bodyGrid = new Slick.Grid(this.$gridDiv, dataView, columns,
+                                   gridOptions);
+
+    this.$searchBar.on('input', function(e) {
+      dataView.refresh();
+
+      // show a message when no results are found
+      if (scope.bodyGrid.getDataLength() === 0 &&
+          scope.$searchBar.val() !== '') {
+        scope.$searchBar.attr('title', 'No results found!');
```

Comment:
Putting this here so I don't forget. In Chrome76 && FF68 the sidebar width increases by about 5-10 px. When the "No results found!" warning goes away, the sidebar resizes to its original size. I haven't tracked down _why_ yet, but its a bit jarring when it happens.

Reply:
I wasn't able to reproduce this, are you using Emperor as a QIIME2 plugin?

Reply:
> are you using Emperor as a QIIME2 plugin?

I am, here is a screenshot via q2-emperor (in Firefox):

![Peek 2019-10-03 13-34](https://user-images.githubusercontent.com/274668/66162125-95cb2980-e5e2-11e9-99fd-a3d8d79e77c2.gif)

I _think_ a scrollbar is being added when the modal display pops up.

Here is a screenshot of a standalone HTML viz, in Google Chrome:

![Peek 2019-10-03 13-46](https://user-images.githubusercontent.com/274668/66163016-4c7bd980-e5e4-11e9-8895-38d012c512cc.gif)

Reply:
Thanks for the screenshots, I'll look into this. Not sure what might be going on.

Reply:
Just went through this I installed a development version of Emperor in my QIIME2 environment and generated new QZVs. However I wasn't able to get this to work neither in Firefox nor in Chrome.

# Firefox:
![missing-thing](https://user-images.githubusercontent.com/375307/66225521-7806ce80-e68d-11e9-9fb4-76105a2a8c0d.gif)


# Chrome
(also @antgonza this is how the menus look now when the application starts, search bar is hidden until a category is selected):

![missing-thing-chrome](https://user-images.githubusercontent.com/375307/66225511-74734780-e68d-11e9-8e0a-321d3345e7ce.gif)

Reply:
Hmm, maybe try clicking around and doing other things after the "no results found" popup?

Reply:
I've tried this but have not seen anythign odd happen. I'll check in some of my colleague's machines. Here's a link to [a test artifact](https://view.qiime2.org/visualization/?type=html&src=https%3A%2F%2Fdl.dropbox.com%2Fs%2Ferlv7f0i1qrddlz%2Fpr-735%3Fdl%3D1) in case someone else is able to test this.

Reply:
Thanks @ElDeveloper - that linked viz has the shifting issue for me when I open it it both Chrome and Firefox. Also, in Chrome I noticed this (bottom right corner):

![image](https://user-images.githubusercontent.com/274668/66329605-a2ea6000-e8e3-11e9-805c-7c0fd93743b3.png)

Reply:
Thanks for catching that! I'll look into this. Seems like the tooltip declaration is leaking between the different views. Also, maybe this is a Linux-specific issue.

Reply:
I just checked on a mac, the pixel bouncing doesn't appear to happen there, but I am running into the modal/tooltip issue there, too:

<img width="661" alt="Untitled" src="https://user-images.githubusercontent.com/274668/66352742-2a9c9280-e915-11e9-89d3-27fa227d4c6d.png">

The popup appears to stay fixed in absolute position when the window is resized.

Reply:
Alright, it took me a little while to setup a linux dev environment but I figured this one out. Should be fixed in the current branch, and for your convenience I've uploaded a new artifact [here](https://view.qiime2.org/visualization/?type=html&src=https%3A%2F%2Fdl.dropbox.com%2Fs%2Fw8n1gs2h698g0kv%2Fpr-735-take2%3Fdl%3D1).

Reply:
Thanks @ElDeveloper, sorry this is a pain. I am still seeing some strange positioning for the "No Results Found!" window:

![Peek 2019-10-09 16-45](https://user-images.githubusercontent.com/274668/66528367-6de13780-eab4-11e9-9054-822197b1bc04.gif)

When the dev console is opened you will notice the "No Results Found!" window moves up above the console. When it moves up, it also moves over to the left 10-15px. When it does that, it "pushes" the main emperor view window over, then the "No Results Found!" window re-centers itself - but the main view window _doesn't resize_ then.

Reply:
Resolving, since this seems to be a bit of an edge case. I suspect this is mostly due to the modal library used.